metadata-db-app/src/cmr/metadata_db/data/oracle/concepts.clj

```diff
@@ -224,6 +224,80 @@
+  Returns nil if valid and an error response if invalid."
+  [db concept]
+  (let [variable-concept-id (get-in concept [:extra-fields :variable-concept-id])
+        associated-concept-id (get-in concept [:extra-fields :associated-concept-id])]
+    (when (and variable-concept-id associated-concept-id))
+      ;; Get all the other variables sharing the same name; not deleted
+      ;; and are associated with the same collection.
+      (let [stmt [(format "select va.variable_concept_id
+                           from   cmr_variable_associations va, 
+                                  (select concept_id, max(revision_id) maxrev
+                                   from cmr_variable_associations group by concept_id) latestva,
+                                  cmr_variables v
+                           where  va.revision_id = latestva.maxrev
+                           and    va.concept_id = latestva.concept_id
+                           and    va.deleted = 0 
+                           and    va.variable_concept_id = v.concept_id
+                           and    va.variable_concept_id != '%s'
+                           and    va.associated_concept_id = '%s'
+                           and    v.variable_name in (select variable_name
+                                                      from   cmr_variables
+                                                      where  concept_id = '%s')"
+                          variable-concept-id associated-concept-id variable-concept-id)]
+            result (first (su/query db stmt))
+            v-concept-id-same-name (:variable_concept_id result)]
+        (when v-concept-id-same-name
+          {:error :collection-associated-with-variable-same-name
+           :error-message (format (str "Variable [%s] and collection [%s] can not be associated "
+                                       "because the collection is already associated with another variable [%s] with same name.")
+                                  variable-concept-id associated-concept-id v-concept-id-same-name)}))))
+
+(defn validate-variable-not-associated-with-another-collection
+  "Validates that variable-concept-id in the concept is not associated with another collection.
+  Returns nil if valid and an error response if invalid."
+  [db concept]
+  (let [variable-concept-id (get-in concept [:extra-fields :variable-concept-id])
+        associated-concept-id (get-in concept [:extra-fields :associated-concept-id])]
+    (when (and variable-concept-id associated-concept-id)
+      ;;Get all the same variables that are not deleted and are associated with other collections.
+      (let [stmt [(format "select va.associated_concept_id
+                           from   cmr_variable_associations va,
+                                  (select concept_id, max(revision_id) maxrev
+                                   from cmr_variable_associations group by concept_id) latestva
+                           where  va.revision_id = latestva.maxrev
```

Comment:
This logic is wrong. the variable association revision-ids do not need to match.

Reply:
You should add a test to cover this case: where a variable is associated with a collection twice, and you then want to associate the variable to a different collection. The test should fail. With the current implementation, the test will pass.

Reply:
I tested this numerous times with demo.  It worked fine. I needed to have the revision-id to have the latest so that I can check if the variable association is deleted. because only when the latest revision is deleted, the association is deleted.

Reply:
ok. test it this way:
1) associate variable A with collection A
2) dissociate variable A with collection A
3) associate variable A with collection A
4) associate variable A with collection B

I am not certain, but I think that will create a variable association with revision id greater than 1.

Reply:
Never mind. I think you are right. I was misreading the query. Don't really have time to review this in detail. Too much going on right now.